### PR TITLE
EntropyReader Class

### PR DESCRIPTION
**Pull request by RanpengLi**
I opened up this pull request to test and fix some issues in my last pull request (#5334 Add EntropyReader Class). The previous one couldn't compile on the testers, but it could on my up-to-date local version. It also seems to have some strange indents. 

* [X] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### CODE REVIEW

File: include/aspect/material_model/utilities.h [hunk 1]
```diff
@@ -259,6 +266,82 @@ namespace aspect
                           const bool interpol,
                           const MPI_Comm &comm);
         };
+
+        /**
+        * This class reads in an entropy-pressure material table and looks up material
+        * proporties for the given entropy and pressure.
```
Comment (gassmoeller) [inline include/aspect/material_model/utilities.h:272]:
```suggestion
        * properties for the given entropy and pressure.
```

File: include/aspect/material_model/utilities.h [hunk 2]
```diff
@@ -259,6 +266,82 @@ namespace aspect
...
+
+            /**
+             * Gets density gradient for a given entropy and pressure.
+             */
+            Tensor<1, 2>
+            density_gradient(const double entropy,
+                             const double pressure) const;
+
+          private:
+            std::unique_ptr<Utilities::StructuredDataLookup<2>> material_lookup;
```
Comment (gassmoeller) [inline include/aspect/material_model/utilities.h:343]:
```suggestion
/**
 * The StucturedDataLookup object that stores the material data.
 */
            std::unique_ptr<Utilities::StructuredDataLookup<2>> material_lookup;
```

File: include/aspect/material_model/utilities.h [hunk 3]
```diff
@@ -259,6 +266,82 @@ namespace aspect
...
+            /**
+             * Read the table.
+             */
+            void
+            initialize(const MPI_Comm comm,
+                       const std::string data_directory,
+                       const std::string material_file_name);
+
+            /**
+             * Reads the specific heat for a given entropy and pressure.
```
Comment (gassmoeller) [inline include/aspect/material_model/utilities.h:287]: `Reads` is maybe not the correct work. What about:
```suggestion
             * Returns the specific heat for a given entropy and pressure.
```
Same for all the other functions below.

File: source/material_model/utilities.cc [hunk 4]
```diff
@@ -786,6 +788,100 @@ namespace aspect
...
+          return thermal_expansivity;
+        }
+
+
+
+        double
+        EntropyReader::compressibility(const double entropy,
+                                       const double pressure) const
+        {
+          const double compressibility = material_lookup->get_data({entropy,pressure}, 2);
```
Comment (gassmoeller) [inline source/material_model/utilities.cc:840]: This seems not correct. The `thermal_expansivity` above is also listed as column 2. Which one is it?

File: source/postprocess/command.cc [hunk 5]
```diff
@@ -29,7 +29,7 @@ namespace aspect
     Command<dim>::execute (TableHandler &)
     {
       if (on_all_processes ||
-          (Utilities::MPI::this_mpi_process(this->get_mpi_communicator()) == 0))
```
Comment (gassmoeller) [inline source/postprocess/command.cc:32]: Are these dealii:: prefixes here and in the other instances still necessary? I though the `using` statements had fixed that?

**Review by gassmoeller — COMMENTED** [review]
Close, just a few comments left.

**Comment by RanpengLi** [comment]
Thank you! I have addressed the comments and squashed the previous commits into one.

**Review by gassmoeller — APPROVED** [review]
Looks good, thank you.

**Pull request merged**